facade/src/test/java/org/jboss/pnc/facade/providers/SCMRepositoryProviderTest.java

```diff
@@ -0,0 +1,292 @@
+import org.jboss.pnc.spi.datastore.repositories.RepositoryConfigurationRepository;
+import org.jboss.pnc.spi.datastore.repositories.api.Repository;
+import org.jboss.pnc.spi.notifications.Notifier;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.MockitoJUnitRunner;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class SCMRepositoryProviderTest extends AbstractProviderTest<RepositoryConfiguration> {
+
+    @Mock
+    private RepositoryConfigurationRepository repository;
+
+    @Mock
+    private Notifier notifier;
+
+    @Mock
+    private Configuration configuration;
+
+    @Mock
+    private ScmModuleConfig scmModuleConfig;
+
+    @InjectMocks
+    private SCMRepositoryProviderImpl provider;
+
+    private RepositoryConfiguration mock = createNewRepositoryConfiguration(
+            "http://external.sh", "git+ssh://internal.sh", true);
+
+    private RepositoryConfiguration mockSecond = createNewRepositoryConfiguration(
+            "http://external2.sh", "git+ssh://internale.sh", false);
+
+    @Before
+    public void setup() {
+
+        List<RepositoryConfiguration> list = new ArrayList<>();
+
+        list.add(mock);
+        list.add(mockSecond);
+
+        list.add(createNewRepositoryConfiguration(
+                "https://" + UUID.randomUUID().toString() +".ca",
+                "git+ssh://" + UUID.randomUUID().toString() + ".eu",
+                true));
+
+        fillRepository(list);
+    }
+
+    @Override
+    protected AbstractProvider provider() {
+        return provider;
+    }
+
+    @Override
+    protected Repository<RepositoryConfiguration, Integer> repository() {
+        return repository;
+    }
+
+    @Test
+    public void testStoreNewRepositoryWithoutId() {
+        // when
+        SCMRepository toCreate = createNewSCMRepository(
+                "https://" + UUID.randomUUID().toString() +".ca",
+                "git+ssh://" + UUID.randomUUID().toString() + ".eu",
+                true,
+                null);
+
+        SCMRepository created = provider.store(toCreate);
+
+        // then
+        assertThat(created).isNotNull();
+        assertThat(created.getId()).isNotNull().isNotEmpty();
+        assertThat(created.getInternalUrl()).isEqualTo(toCreate.getInternalUrl());
+        assertThat(created.getExternalUrl()).isEqualTo(toCreate.getExternalUrl());
+        assertThat(created.getPreBuildSyncEnabled()).isEqualTo(toCreate.getPreBuildSyncEnabled());
```

Comment:
SCMRepositoryProvider.store should be disabled IMO

Reply:
I'll double check what we use for BPM to create the entries. Can't find it now :(

Reply:
@janinko Checking from Maitai, we send a request to Maitai for the repository configuration creation, and Maitai calls `POST pncUrl/repository-configurations` to create a new repository configuration endpoint in the database.

What's the equivalent of this in PNC 2.0? I don't see it.

Reply:
There is none ATM IIRC.

Reply:
@janinko  Given this information, I wouldn't remove it because we'll probably use it in the future. WDYT?

Reply:
@janinko ping :)

Reply:
yeah, you are right